src/parse/mod.rs

```diff
@@ -328,7 +324,18 @@ impl<I: Iterator<Item = Lexeme>> Parser<I> {
         mem::replace(&mut self.next, tmp);
     }
     fn lex_error(&mut self, err: CompileError) {
-        self.pending.push_back(Err(err));
+        self.error_handler.push_back(err);
+    }
+    pub fn collect_results(self) -> (Vec<Locatable<Declaration>>, Vec<CompileError>) {
```

Comment:
This function can be removed in favor of [partition](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.partition). The call in main would look like `Parser::new(...).partition(|res| res.is_err())`.

Reply:
We decided against this because when we add warnings in the next PR we couldn't partition into 3 collections.